### ISSUE TITLE
kvm (virt-install)

### ISSUE THREAD

**Issue by atonKin86** [home-assistant/home-assistant.github.io#29695]
### Feedback

After creating HA VM you suggest to reach Home Assistant on http://homeassistant.local:8123/
BUT there is no network configuration on virt-install command.

### URL

https://www.home-assistant.io/installation/linux/

### Version

2023.11.0

**Comment by rimmer** [comment]
This is still relevant. I spent hours to figure out the solution and .. there are several solutions depending on your goals and the setup. 
### The good.
Well, good enough for me. It is about using direct network interface with [MacVTap](https://virt.kernelnewbies.org/MacVTap) driver (that kinda should be installed on your system, at least it was on Fedora CoreOS). 
1. Edit your VM with `virsh edit haos`
2. Find `<interface ...>` under `<devices>` section
3. Should look like (where `enp0s31f6` is your **wired** connection).
```xml
    <interface type='direct'>
      <!-- some other stuff -->
      <source dev='enp0s31f6' mode='bridge'/>
      <model type='virtio'/>
      <!-- some other stuff -->
    </interface>
```
4. Save, reload

A one liner might be described in [this Gist](https://gist.github.com/smurugap/163b3e2be7676a46c835339f8ba0710f), but I haven't tried yet. 
TLDR; add `--network type=direct,source=enp6s0f0,source_mode=bridge,model=virtio` to the `virt-install`. 

#### Pros
- No need to setup bridge upfront

#### Cons
- MacVTap should be there
- Doesn't work with WiFi, believe me, I tried. 

### The bad
Same as above, but set up a bridge upfront. If you aren't a 10 years experience devops, this is quite troublesome, just google it (sorta "set up bridge for libvirt"). 
You still have to add `--network br0` to `virt-install`

#### Pros
- No need for MacVTap
- It is ~~purely~~ [documented](https://docs.redhat.com/en/documentation/red_hat_enterprise_linux/7/html/virtualization_deployment_and_administration_guide/sect-guest_virtual_machine_installation_overview-creating_guests_with_virt_install#sect-Guest_virtual_machine_installation_overview-virt_install-network_installation)
- nerdness points +2

#### Cons
- More manual work that one has to replicate and support if provisioning. 

### The ugly
Just do port forwarding. Though, for me it was even more troublesome, I couldn't get it working. In fact I would prefer this to be working because I'd prefer WiFi connection to the wired one, but it didn't so now my home server jeopardizing my relationship by standing wired to the router. 

#### Pros
- No need for MacVTap
- Supposed to be working with WiFi (which is huge for someone)

#### Cons
- HA is not a separate instance in your network.
- Perhaps something else which I can't think of right now.

### Suggestion
I would suggest at least adding `--network type=direct,source=enp6s0f0,source_mode=bridge,model=virtio` to the `virt-install` to the manual (verifying it prior ofc), at most add more info and maybe a better solution.
